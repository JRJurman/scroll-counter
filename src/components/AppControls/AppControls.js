import { registerHtml, useUrlParams } from 'tram-one'
import './AppControls.scss'

const html = registerHtml()

export default (props, children) => {
  const { min = -100, max = 100, start = 0 } = useUrlParams()
  const onChangeMin = (event) => {
    window.location.search = `?min=${event.target.value}&max=${max}&start=${start}`
  }
  const onChangeMax = (event) => {
    window.location.search = `?min=${min}&max=${event.target.value}&start=${start}`
  }
  const onChangeStart = (event) => {
    window.location.search = `?min=${min}&max=${max}&start=${event.target.value}`
  }
  return html`
    <div class="AppControls">
      Min: <input type="number" size="4" placeholder="-100" value=${min} onchange=${onChangeMin} />
      Max: <input type="number" size="4" placeholder="100" value=${max} onchange=${onChangeMax} />
      Start: <input type="number" size="4" placeholder="0" value=${start} onchange=${onChangeStart} />
    </div>
  `
}
