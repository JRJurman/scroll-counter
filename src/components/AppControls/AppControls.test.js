import AppControls from './AppControls'

describe('AppControls', () => {
  describe('with default props', () => {
    const wrapper = AppControls()

    it('should match snapshot', () => {
      expect(wrapper.outerHTML).toMatchSnapshot()
    })
  })
})
