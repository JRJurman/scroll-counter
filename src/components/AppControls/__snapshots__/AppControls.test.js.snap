// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppControls with default props should match snapshot 1`] = `
"<div class=\\"AppControls\\">
      Min: <input type=\\"number\\" size=\\"4\\" placeholder=\\"-100\\" value=\\"-100\\">
      Max: <input type=\\"number\\" size=\\"4\\" placeholder=\\"100\\" value=\\"100\\">
      Start: <input type=\\"number\\" size=\\"4\\" placeholder=\\"0\\" value=\\"0\\"></div>"
`;
